packages/statusbar/src/defaults/kernelStatus.tsx

```diff
@@ -183,8 +181,11 @@ export namespace KernelStatus {
       const oldState = this._getAllState();
       const { newValue } = change;
       if (newValue !== null) {
-        this._kernelStatus = newValue.status;
-        this._kernelName = newValue.model.name.toLowerCase();
+        newValue.getSpec().then(spec => {
+          // sync setting of status and display name
+          this._kernelStatus = newValue.status;
+          this._kernelName = spec.display_name;
```

Comment:
We should probably call `_triggerChange` here as well, since this will be updated after the `_triggerChange` called below.  Maybe the below `_triggerChanged` should be moved inside of the `else` block as well.

Reply:
A shame that this function is async, but I see no way around it.

Reply:
done